### ISSUE TITLE
ffmpeg-5.0 causes owntone to restart on audio playback 

### ISSUE THREAD

**Issue by richjn** [owntone/owntone-server#1396]
Owntone starts / scans library but on playing audio restarts itself. 

owntone-server 28.2 bison_parser1 branch.

Arch Linux custom / non-distro ffmpeg

Compile the recently released ffmpeg-5.0 (usual soname bumps) then re-compile owntone.
Try to play audio
Neither LOG or DEBUG show any useful output that I can see.
Reverting to ffmpeg-4.4.1 and re-compiling owntone restores functionality.

**Comment by ejurgensen** [comment]
Yes, I could reproduce. Gotta give it to ffmpeg, they can be counted on to break their API.

As you can see I have made some changes that I think should fix it. There is one change that I'm concerned could break something (not removing the AVFMT_NOFILE flag), so I will test a bit more. If you can also test the branch it would be great.

**Comment by ejurgensen** [comment]
I guess I should add that the branch with the fix is based on master, so that's with antlr as parser. You can merge the branches locally if you want to stick with bison/flex.

**Comment by richjn** [comment]
Yes, I had wanted to stick with the bison_parser1 branch and simply patched the transcode.c changes (I have also merge @chme web-next) I will need to keep an eye out if you make further changes.
Happy to report everything seems to be working fine here, I will update if I notice anything.

**Comment by ejurgensen** [comment]
I've merged a modified fix that should also work with older versions of ffmpeg. Thanks for reporting and testing.
